lemur/plugins/lemur_entrust/plugin.py

```diff
@@ -313,10 +313,88 @@ class EntrustSourcePlugin(SourcePlugin):
+        required_vars = [
+            "ENTRUST_API_CERT",
+            "ENTRUST_API_KEY",
+            "ENTRUST_API_USER",
+            "ENTRUST_API_PASS",
+            "ENTRUST_URL",
+            "ENTRUST_ROOT",
+            "ENTRUST_NAME",
+            "ENTRUST_EMAIL",
+            "ENTRUST_PHONE",
+        ]
+        validate_conf(current_app, required_vars)
+
+        self.session = requests.Session()
+        cert_file = current_app.config.get("ENTRUST_API_CERT")
+        key_file = current_app.config.get("ENTRUST_API_KEY")
+        user = current_app.config.get("ENTRUST_API_USER")
+        password = current_app.config.get("ENTRUST_API_PASS")
+        self.session.cert = (cert_file, key_file)
+        self.session.auth = (user, password)
+        self.session.hooks = dict(response=log_status_code)
+        super(EntrustSourcePlugin, self).__init__(*args, **kwargs)
 
     def get_certificates(self, options, **kwargs):
-        # Not needed for ENTRUST
-        raise NotImplementedError("Not implemented\n", self, options, **kwargs)
+        """ Fetch all Entrust certificates """
+        base_url = current_app.config.get("ENTRUST_URL")
+        host = base_url.replace('/enterprise/v2', '')
+
+        get_url = f"{base_url}/certificates"
+        certs = []
+        offset = 0
+        while True:
+            response = self.session.get(get_url,
+                 params={
+                     "status": "ACTIVE",
+                     "isThirdParty": "false",
+                     "fields": "uri,dn",
+                     "offset": offset
+                 }
+            )
+            try:
+                data = json.loads(response.content)
+            except ValueError:
+                # catch an empty jason object here
+                data = {'response': 'No detailed message'}
+            status_code = response.status_code
+            if status_code > 399:
+                raise Exception(f"ENTRUST error: {status_code}\n{data['errors']}")
+            # current_app.logger.info(f"recevied: {data['summary']}")
+            for c in data["certificates"]:
+                download_url = "{0}{1}".format(
+                    host, c["uri"]
+                )
+                cert_response = self.session.get(download_url)
+                certificate = json.loads(cert_response.content)
+                # current_app.logger.info(f"Result: {certificate}")
+                # normalize serial
+                serial = str(int(certificate["serialNumber"], 16))
+                cert = {
+                    "body": certificate["endEntityCert"],
+                    "serial": serial,
+                    "external_id": str(certificate["trackingId"]),
+                }
+                certs.append(cert)
+            if data["summary"]["limit"] * offset >= data["summary"]["total"]:
+                break
+            else:
+                offset += 1
+        current_app.logger.info(f"Result: {certs}")
```

Comment:
wondering if logging all the certs is a good idea, or rather the total number of discovered certs?

Reply:
I'll take this out